### PR TITLE
Exclude all TPLs from Doxygen

### PR DESCRIPTION
**Pull request by davidbeckingsale**

### CODE REVIEW

**Comment by davidbeckingsale** [comment]
@mcfadden8 - this fixes the failing ReadTheDocs build.

File: docs/doxygen/Doxyfile.in [hunk 1]
```diff
@@ -507,7 +507,8 @@ EXCLUDE_SYMLINKS       = NO
 
 EXCLUDE_PATTERNS       = */tpl/simpool/* \
                          */tpl/judy/* \
-                         */tpl/conduit/* 
+                         */tpl/conduit/* \
+                         */tpl/cxxopts/*
```
Comment (mcfadden8) [inline docs/doxygen/Doxyfile.in]: Looks good!

Should we replace all of these with:
```suggestion
                         */tpl/* 
```
That way, we won't have to remember to explicitly add other TPLS (json is coming) to this exclusion list

Comment (davidbeckingsale) [reply docs/doxygen/Doxyfile.in]: Let's try it and see what happens 😄

Comment (mcfadden8) [reply docs/doxygen/Doxyfile.in]: If this works, I suggest replacing
```
EXCLUDE_PATTERNS  = */tpl/simpool/* \
                        */tpl/judy/* \
                        */tpl/conduit/* \
                        */tpl/cxxopts/*
```
with
```
EXCLUDE_PATTERNS = */tpl/*
```

I wish there was a way I could use the sugguestion button for this...

**Pull request merged**